src/app/replayer/replayer.ml

```diff
@@ -4,57 +4,71 @@ open Core
 open Async
 open Coda_base
 
-(* identify a target block B to replay to, either with
+(* identify a target block B to replay to, by indicating its
+   state hash
 
-   1)  target state hash  = protocol state hash in B
-       target proof       = blockchain SNARK proof of the protocol state in B
+   from B, we choose a successor block, and obtain its "staking"
+   and "next" epoch ledger hashes
```

Comment:
Could we do this the other way you suggested? i.e.,the input takes a state hash that is the successor of the block being forked off of. The epoch data is then obtained from epoch_data_ids corresponding to the input state hash. The commands for genesis ledger will be from the parent of the input state hash.
This will ensure the replayer has always two state hashes to deal with. With current implementation, there could be multiple successor states and the tool would have to choose one of them which is not trivial- if the successors are all on the same slot then it could pick one and  any one if the successors happen to be on different slots then it should pick the one on the earliest slot provided it is the accepted by the majority of nodes